docs/src/tutorials/symbmc.md

```diff
@@ -0,0 +1,657 @@
+
+For example, given
+```tla
+VARIABLE 
+  \* @type: Int;
+  x,
+  \* @type: Int;
+  y
+
+Init == x \in 3..5 /\ y = 2 
+```
+we see that `S = [x: Z, y: Z]` and `S0 = { [x |-> 3, y |-> 2], [x |-> 4, y |-> 2], [x |-> 5, y |-> 2] }`.
+
+### Transitions
+Similar to `S0`, `->` is derived from `S` and `Next`. If `S0` is a single-argument predicate `S0 \in [S -> BOOLEAN]`, then `->` is a relation `-> \in [S^2 -> BOOLEAN]`. 
+`->(s1,s2)` is the formula obtained by replacing all occurrences of variable names `Ai` in `Next` with the values `s1.Ai`, and all occurrences of `Ai'` with `s2.Ai`.
+By the same principle of subset-predicate equivalence, we can treat `->` as a subset of `S^2`.
+As mentioned in the notation section, it is generally more convenient to use the infix notation
+`s1 -> s2` over `->(s1, s2)`. We say that a state `s2` is a _successor_ of the state `s1` if `s1 -> s2`. 
+
+For example, given
+```tla
+VARIABLE 
+  \* @type: Int;
+  x,
+  \* @type: Int;
+  y
+
+Init == x \in 3..5 /\ y = 2 
+
+Next == x' \in { x, x + 1 } /\ UNCHANGED y
+```
+One can deduce, for any state `[x |-> a, y |-> b] \in S`, that it has two successors: `[x |-> a + 1, y |-> b]` and `[x |-> a, y |-> b]` because the following relations hold `[x |-> a, y |-> b]-> [x |-> a + 1, y |-> b]` and `[x |-> a, y |-> b] -> [x |-> a, y |-> b]`.
+
+### Reachable states
+Using the above definitions, we can define the set of states reachable in exactly `k`-steps, for `k \in N`, denoted by `R(k)`. We define `R(0) = S0` and for each `k \in N`,
+```tla
+R(k+1) := { t \in S: \E s \in R(k): s -> t }
+```
+
+Similarly, we can define the set of states reachable in _at most_ `k`-steps, denoted `r(k)`, for `k \in N` by
+```tla
+r(k) = R(0) \union R(1) \union ... \union R(k-1) \union R(k)
+```
+
+Finally, we define the et of all _reachable_ states, `R`, as the (infinite) union of all `r(k)`, over `k \in N`:
+```tla
+R = r(0) \union r(1) \union ...
+```
+
+For example, given
+```tla
+VARIABLE 
+  \* @type: Int;
+  x,
+  \* @type: Int;
+  y
+
+Init == x \in 1..3 /\ y = 2
+
+Next == x' = x + 1 /\ UNCHANGED y
+```
+we can deduce: 
+```tla
+R(0) = r(0) = S0 = {[x: 1, y: 2],[x: 2, y: 2],[x: 3, y: 2]}
+
+R(1) = {[x: 2, y: 2], [x: 3, y: 2], [x: 4, y: 2]}
+r(1) = {[x: 1, y: 2], [x: 2, y: 2], [x: 3, y: 2], [x: 4, y: 2]}
+
+R(2) = {[x: 3, y: 2], [x: 4, y: 2], [x: 5, y: 2]}
+r(2) = {[x: 1, y: 2], [x: 2, y: 2], [x: 3, y: 2], [x: 4, y: 2], [x: 5, y: 2]}
+```
+and so on. We can express this compactly as:
+```
+R(i) = [x: (i+1)..(i+3), y: {2}],
+r(i) = [x: 1..(i+3)    , y: {2}],
+R    = [x: N           , y: {2}]`
+```
+
+### Finite diameters
+We say that a transition system has a _finite diameter_, if there exists a `k \in N`, such that `R = r(k)`.
```

Comment:
Why are finite diameters relevant in this discussion? Every finite transition system has a finite diameter that is equal to the number of its states in the worst case.

Reply:
State spaces can be infinite, e.g.
```
Init == x = 0
Next == x' = x + 1
```
The point is that MC will only terminate if `diam< inf`, and BMC == MC if `k > diam`.